industrial_robot_client/src/joint_trajectory_action.cpp

```diff
@@ -216,9 +210,12 @@ void JointTrajectoryAction::cancelCB(JointTractoryActionServer::GoalHandle & gh)
 
 void JointTrajectoryAction::controllerStateCB(const control_msgs::FollowJointTrajectoryFeedbackConstPtr &msg)
 {
-  //ROS_DEBUG("Checking controller state feedback");
+  ROS_DEBUG("Checking controller state feedback");
   last_trajectory_state_ = msg;
-  trajectory_state_recvd_ = true;
+  controller_alive_ = true;
+
+  watchdog_timer_.stop();
+  watchdog_timer_.start();
```

Comment:
that feels like using a side-effect, stop start is a bit clumsy but quite legible, but I have no problem with being overvoted